devicetypes/smartthings/qubino-flush-relay.src/qubino-flush-relay.groovy

```diff
@@ -0,0 +1,436 @@
+	} else {
+		return false
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private syncConfiguration() {
+	def commands = []
+	parameterMap.each {
+		if (state.currentPreferencesState."$it.key".status == "syncPending") {
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		} else if (state.currentPreferencesState."$it.key".status == "disablePending") {
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		}
+	}
+	sendHubCommand(commands)
+}
+
+private getParameterMap() {[
+	[
+		name: "Input 1 switch type", key: "input1SwitchType", type: "boolean",
+		parameterNumber: 1, size: 1, defaultValue: 1,
+		optionInactive: 0, inactiveDescription: "mono-stable switch type (push button)",
+		optionActive: 1, activeDescription: " bi-stable switch type ",
+		description: "Defines the type of switch connected to Input 1 "
+	],
+	[
+		name: "Input 2 contact type", key: "input2ContactType", type: "boolean",
+		parameterNumber: 2, size: 1, defaultValue: 0,
+		optionInactive: 0, inactiveDescription: "NO (normally open) input type",
+		optionActive: 1, activeDescription: "1 -  NC (normally close) input type",
+		description: "Defines the type of switch connected to Input 2 "
+	],
+	[
+		name: "Saving the state of the relay after a power failure", key: "savingTheStateOfTheRelayAfterAPowerFailure", type: "boolean",
+		parameterNumber: 30, size: 1, defaultValue: 0,
+		optionInactive: 0, inactiveDescription: "Flush 1D relay module saves its state before power failure (it returns to the last position saved before a power failure)",
+		optionActive: 1, activeDescription: "Flush 1D does not save on/off status and does not restore it after a power failure, it remains off.",
+		description: "Support saving the state of the relay after a power failure "
+	],
+	[
+		name: "Output Switch selection", key: "outputSwitchSelection", type: "boolean",
+		parameterNumber: 63, size: 1, defaultValue: 0,
+		optionInactive: 0, inactiveDescription: "When system is turned off the output is 0V (NC).",
+		optionActive: 1, activeDescription: "When system is turned off the output is 230V or 24V (NO).",
+		description: "Set the type of device connected to the output"
+	],
+	//Only for Qubino 1 Relay
+	[ 
+		name: "Input 3 contact type", key: "input3ContactType", type: "boolean",
```

Comment:
So, if we're not surfacing the switches attached to input 1 and 3 at all anyway, why are these preferences here? 

I thought we were just going to add them as simple switch (or contact sensor) child devices?